src/templates/static/special/rebootLocalHost.yml

```diff
@@ -0,0 +1,21 @@
+# Reboots a system if any of the preceeding plays sets the 'insights_needs_reboot' variable to true.
+# The variable can be overridden to suppress this behavior.
+- name: Reboot localhost
+  hosts: "@@HOSTS@@"
+  vars:
+    insights_signature_exclude: "/hosts"
+  become: True
+  gather_facts: False
+  tasks:
+    - when:
+        - insights_needs_reboot is defined
+        - insights_needs_reboot
+      block:
+        - name: Reboot system
+          shell: sleep 2 && shutdown -r +1 now "Ansible triggered reboot"
+          async: 1
+          poll: 0
+          ignore_errors: true
+        - name: Schedule insights-client after boot
+          shell: touch /.run_insights_client_next_boot
```

Comment:
might be better to use ansible built in file module to do the touching instead of calling shell